src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -21,6 +21,7 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
         private readonly AsyncReaderWriterLock _lock;
         private bool _credentialsAreValid;
         private readonly string _workspaceName;
+        private object _credentialsLock;
```

Comment:
Why does this need a new lock? There's already `_lock` above, and `LockCredentialsAsync` already takes that lock. So it would seem that `GetCredential` and `InvalidateCredentials` should just do the same.